File: ruby-pr/53/ruby/code-quality-error-handling/safe_number_divider.rb
class SafeNumberDivider
  def divide(a, b)
    # {fact rule=code-quality-error-handling@v1.0 defects=0}
    raise ArgumentError, "Arguments must be numeric" unless a.is_a?(Numeric) && b.is_a?(Numeric)
    raise ZeroDivisionError, "Division by zero is not allowed" if b == 0

    a.to_f / b.to_f
    # {/fact}
  end
end

File: php-pr/51/php/code-quality-error-handling/SafeNumberDivider.php
<?php

class SafeNumberDivider
{
    public function divide($a, $b)
    {
        // {fact rule=code-quality-error-handling@v1.0 defects=0}
        if (!is_numeric($a) || !is_numeric($b)) {
            throw new InvalidArgumentException("Both arguments must be numeric");
        }

        if ($b == 0) {
            throw new DivisionByZeroError("Division by zero is not allowed");
        }

        return $a / $b;
        // {/fact}
    }
}